Formula/libcoap.rb

```diff
@@ -11,6 +11,8 @@ class Libcoap < Formula
     sha256 "a68df19a4ca87c677173c14b534848592bb35e46a715ca066bcd114f8c735236" => :high_sierra
   end
 
+  option "with-examples", "Will build the included POSIX examples"
```

Comment:
We don't allow options in homebrew-core.

Reply:
Ah, my bad. I must have missed the note in the Formula Cookbook. I see now.

Shame it's not allowed, this would allow much more versatile usage of brew install without having to tap other taps. But I won't discuss it here.

**Thanks for the feedback.**

Reply:
> Shame it's not allowed, this would allow much more versatile usage of brew install without having to tap other taps. But I won't discuss it here.

In my experience it allows mostly for a much more versatile collection of build failures, triggered by people who found some blog from 2012 that sorta describes their desired install. If you think this change will benefit the majority of Homebrew users you might just want to make it the default though.

Reply:
`coap-client` is definitely a useful tool.

How should this be done then?

Debian provides a [`libcoap2-bin`](https://packages.debian.org/buster/libcoap2-bin) package with all the examples including `coap-client`.
Fedora does the same with [`libcoap-utils`](https://rpmfind.net/linux/RPM/fedora/32/x86_64/l/libcoap-utils-4.2.1-1.fc32.x86_64.html).

Is the right option to create a new `libcoap-bin`/`libcoap-utils` formula that depends on `libcoap` with all the examples its examples?